### PR TITLE
Warrior Pack

### PR DESCRIPTION
**Pull request by Paul-Sml**
Diamsword's Warrior Pack
along with few fixes to my other packs

### CODE REVIEW

**Comment by Paul-Sml** [comment]
just added vfx sfx

File: src/main/java/thePackmaster/cardmodifiers/warriorpack/FastDamage.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import com.evacipated.cardcrawl.mod.stslib.damagemods.AbstractDamageModifier;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.localization.CardStrings;
+import thePackmaster.SpireAnniversary5Mod;
+
+//Currently serves as a block bypass for Rapier
+public class FastDamage extends AbstractDamageModifier {
+    public static final String ID = SpireAnniversary5Mod.makeID("FastDamage");
+    public final CardStrings cardStrings = CardCrawlGame.languagePack.getCardStrings(ID);
```
Comment (erasels) [inline src/main/java/thePackmaster/cardmodifiers/warriorpack/FastDamage.java:12]: does nothing?

File: src/main/java/thePackmaster/cardmodifiers/warriorpack/FeralDamage.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.localization.CardStrings;
+import com.megacrit.cardcrawl.monsters.AbstractMonster;
+import com.megacrit.cardcrawl.powers.AbstractPower;
+import thePackmaster.SpireAnniversary5Mod;
+
+public class FeralDamage extends AbstractDamageModifier {
+    public static final String ID = SpireAnniversary5Mod.makeID("FrontDamage");
+    public final CardStrings cardStrings = CardCrawlGame.languagePack.getCardStrings(ID);
```
Comment (erasels) [inline src/main/java/thePackmaster/cardmodifiers/warriorpack/FeralDamage.java:16]: ?

File: src/main/java/thePackmaster/cardmodifiers/warriorpack/FrontDamage.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.cards.DamageInfo;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.localization.CardStrings;
+import thePackmaster.SpireAnniversary5Mod;
+
+public class FrontDamage extends AbstractDamageModifier {
+    public static final String ID = SpireAnniversary5Mod.makeID("FrontDamage");
+    public final CardStrings cardStrings = CardCrawlGame.languagePack.getCardStrings(ID);
```
Comment (erasels) [inline src/main/java/thePackmaster/cardmodifiers/warriorpack/FrontDamage.java:14]: ?

File: src/main/java/thePackmaster/cards/warriorpack/FFBlade.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    public final static String ID = makeID(FFBlade.class.getSimpleName());
+
+    private static final int COST = 1;
+
+    public FFBlade(){
+        super(ID, COST, CardType.ATTACK, CardRarity.RARE, CardTarget.ENEMY);
+        baseDamage = 8;
+        DamageModifierManager.addModifier(this, new FeralDamage());
+        DamageModifierManager.addModifier(this, new FrontDamage());
```
Comment (erasels) [inline src/main/java/thePackmaster/cards/warriorpack/FFBlade.java:28]: Not using inherent?

File: src/main/java/thePackmaster/cards/warriorpack/Fall.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+    public final static String ID = makeID(Fall.class.getSimpleName());
+
+    private static final int COST = 2;
+    private static final int BASE_DAMAGE = 6;
+
+    public Fall(){
+        super(ID, COST, CardType.ATTACK, CardRarity.UNCOMMON, CardTarget.ENEMY);
+        baseDamage = BASE_DAMAGE;
+        DamageModifierManager.addModifier(this, new FrontDamage());
```
Comment (erasels) [inline src/main/java/thePackmaster/cards/warriorpack/Fall.java:28]: inherent?

File: src/main/java/thePackmaster/cards/warriorpack/Fall.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        if (!upgraded)
+            this.rawDescription = cardStrings.DESCRIPTION + cardStrings.EXTENDED_DESCRIPTION[0];
+        else
+            this.rawDescription = cardStrings.UPGRADE_DESCRIPTION + cardStrings.EXTENDED_DESCRIPTION[0];
+
+        this.initializeDescription();
+    }
+
+    public void calculateCardDamage(AbstractMonster mo) {
+        super.calculateCardDamage(mo);
```
Comment (erasels) [inline src/main/java/thePackmaster/cards/warriorpack/Fall.java:53]: Doesn#t do the baseDamage calculation here. Not needed? I forget

File: src/main/java/thePackmaster/cards/warriorpack/Rapier.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+    public final static String ID = makeID(Rapier.class.getSimpleName());
+
+    private static final int COST = 0;
+
+    public Rapier(){
+        super(ID, COST, CardType.ATTACK, CardRarity.COMMON, CardTarget.ENEMY);
+        baseDamage = 4;
+        DamageModifierManager.addModifier(this, new FrontDamage());
+        DamageModifierManager.addModifier(this, new FastDamage());
```
Comment (erasels) [inline src/main/java/thePackmaster/cards/warriorpack/Rapier.java:27]: inherent?

File: src/main/resources/anniv5Resources/localization/eng/warriorpack/Cardstrings.json [hunk 8]
```diff
@@ -0,0 +1,44 @@
+{
+  "${ModID}:Sharpening": {
+    "NAME": "Sharpening",
```
Comment (erasels) [inline src/main/resources/anniv5Resources/localization/eng/warriorpack/Cardstrings.json:3]: Please rename, Claw pack already has Sharpen as a mechanic

File: src/main/java/thePackmaster/cards/warriorpack/StormStrike.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+    public final static String ID = makeID(StormStrike.class.getSimpleName());
+
+    private static final int COST = 1;
+
+    public StormStrike(){
+        super(ID, COST, CardType.ATTACK, CardRarity.COMMON, CardTarget.ENEMY);
+        tags.add(CardTags.STRIKE);
+        baseDamage = 6;
+        DamageModifierManager.addModifier(this, new FeralDamage());
```
Comment (erasels) [inline src/main/java/thePackmaster/cards/warriorpack/StormStrike.java:28]: inherent?

File: src/main/java/thePackmaster/cards/warriorpack/WorkHammer.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+public class WorkHammer extends AbstractPackmasterCard {
+
+    public final static String ID = makeID(WorkHammer.class.getSimpleName());
+
+    private static final int COST = 1;
+
+    public WorkHammer(){
+        super(ID, COST, CardType.ATTACK, CardRarity.UNCOMMON, CardTarget.ENEMY);
+        baseDamage = 8;
+        DamageModifierManager.addModifier(this, new FrontDamage());
```
Comment (erasels) [inline src/main/java/thePackmaster/cards/warriorpack/WorkHammer.java:29]: inherent?

File: src/main/resources/anniv5Resources/localization/eng/warriorpack/Cardstrings.json [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+    "DESCRIPTION": "Whenever you play an Attack, increase its damage by !M! this combat."
+  },
+  "${ModID}:WorkHammer": {
+    "NAME": "Work Hammer",
+    "DESCRIPTION": "Deal !D! ${ModID}:Front damage. NL Both you and the enemy gain !B! Block."
```
Comment (erasels) [inline src/main/resources/anniv5Resources/localization/eng/warriorpack/Cardstrings.json:8]: Does not actually describe its effect correctly. Order of operations is important here.
Also very much an iron wave, please rethink this card. Maybe something scaling? Deal FrontDamage, if the enemy has block increase this cards damage by amount this combat?

File: src/main/java/thePackmaster/powers/warriorpack/TurtlePower.java [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+            this.flash();
+            if (amount <= 0)
+                this.addToBot(new RemoveSpecificPowerAction(this.owner, this.owner, this));
+            this.updateDescription();
+        }
+    }
+
+    @Override
+    public List<AbstractDamageModifier> modsToPush(DamageInfo damageInfo, Object o, List<AbstractDamageModifier> list) {
+        return Collections.singletonList(new FeralDamage());
```
Comment (erasels) [inline src/main/java/thePackmaster/powers/warriorpack/TurtlePower.java:53]: Does this remove other modifiers that were in the given list?

File: src/main/resources/anniv5Resources/localization/eng/warriorpack/Cardstrings.json [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+    "NAME": "Chained",
+    "DESCRIPTION": "Gain !B! Block. NL Give ALL enemies !M! Block."
+  },
+  "${ModID}:StormStrike": {
+    "NAME": "Storm Strike",
+    "DESCRIPTION": "Deal !D! ${ModID}:Feral damage. NL Apply !M! Vulnerable."
+  },
+  "${ModID}:WarSong": {
+    "NAME": "War Song",
+    "DESCRIPTION": "Apply !M! Vulnerable. NL Your next Attack deals !${ModID}:m2! additional damage."
```
Comment (erasels) [inline src/main/resources/anniv5Resources/localization/eng/warriorpack/Cardstrings.json:20]: Could probably directly mention vigor here instead

File: src/main/resources/anniv5Resources/localization/eng/warriorpack/Keywordstrings.json [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+      "feral"
+    ],
+    "DESCRIPTION": "Deals #b50% extra damage to enemies with a debuff."
+  },
+  {
+    "PROPER_NAME": "Front damage",
+    "NAMES": [
+      "front"
+    ],
+    "DESCRIPTION": "Deals #b50% extra damage to enemies with #yBlock."
```
Comment (erasels) [inline src/main/resources/anniv5Resources/localization/eng/warriorpack/Keywordstrings.json:14]: Maybe additional is a better word here?

File: src/main/resources/anniv5Resources/localization/eng/warriorpack/Powerstrings.json [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+  "${ModID}:TurtlePower": {
+    "NAME": "Turtle",
+    "DESCRIPTIONS": [
+      "Your next Attack deals Feral damage.",
+      "Your next #b",
+      " Attacks deals Feral damage."
+    ]
+  },
+  "${ModID}:SharpeningPower": {
+    "NAME": "Sharpening",
```
Comment (erasels) [inline src/main/resources/anniv5Resources/localization/eng/warriorpack/Powerstrings.json:11]: Rename please

**Review by erasels — CHANGES_REQUESTED** [review]

File: src/main/java/thePackmaster/cardmodifiers/warriorpack/FastDamage.java (hunk 1 above)
Comment (Paul-Sml) [reply src/main/java/thePackmaster/cardmodifiers/warriorpack/FastDamage.java:12]: Allows Rapier to bypass block ?

File: src/main/resources/anniv5Resources/localization/eng/warriorpack/Cardstrings.json (hunk 11 above)
Comment (Paul-Sml) [reply src/main/resources/anniv5Resources/localization/eng/warriorpack/Cardstrings.json:8]: Iron wave says deal damage then gain block, the player still gains block before. Also your idea overlaps the power I think ? Could do other stuff however it kind of allows chain Front Damage, is it not an acceptable case of iron waving ?
Will try to find smth if it really can't stay the same

File: src/main/java/thePackmaster/cards/warriorpack/Fall.java (hunk 6 above)
Comment (Paul-Sml) [reply src/main/java/thePackmaster/cards/warriorpack/Fall.java:53]: I don't understand ? I've tested with damage modifying powers (vuln/Vigor) it scales

File: src/main/java/thePackmaster/cardmodifiers/warriorpack/FastDamage.java (hunk 1 above)
Comment (erasels) [reply src/main/java/thePackmaster/cardmodifiers/warriorpack/FastDamage.java:12]: cardStrings, i mean

File: src/main/java/thePackmaster/cards/warriorpack/Fall.java (hunk 6 above)
Comment (erasels) [reply src/main/java/thePackmaster/cards/warriorpack/Fall.java:53]: Is it still affected when hovering the monster with the card? If so, does it deal the correct damage?

File: src/main/resources/anniv5Resources/localization/eng/warriorpack/Cardstrings.json (hunk 11 above)
Comment (erasels) [reply src/main/resources/anniv5Resources/localization/eng/warriorpack/Cardstrings.json:8]: Actual effect is more like "Gain Block. Deal front Damage. Give enemy Block." Players needs to know they gain Block before damage to know if they can attack a thorns enemy.

In the end, this is another Ironwave but it does synergize with your pack mechanics enough that you can leave it if you really want it to stay.

Comment (AlexMdle) [reply src/main/resources/anniv5Resources/localization/eng/warriorpack/Cardstrings.json:8]: https://static.wikia.nocookie.net/slay-the-spire/images/9/9e/IronWave.png/revision/latest?cb=20181016210817
Nah, it says block, then damage,

File: src/main/java/thePackmaster/cards/warriorpack/Fall.java (hunk 6 above)
Comment (Paul-Sml) [reply src/main/java/thePackmaster/cards/warriorpack/Fall.java:53]: don't know why it did the form once but not twice

Comment (erasels) [reply src/main/java/thePackmaster/cards/warriorpack/Fall.java:53]: Gotcha, all good then

File: src/main/java/thePackmaster/powers/warriorpack/TurtlePower.java (hunk 12 above)
Comment (Paul-Sml) [reply src/main/java/thePackmaster/powers/warriorpack/TurtlePower.java:53]: Works with in combination with my Front damage so no it doesn't ig ?

**Review by erasels — APPROVED** [review]

**Pull request merged**